packages/main/src/plugin/extension-loader.ts

```diff
@@ -266,7 +266,49 @@ export class ExtensionLoader {
       },
     };
 
-    this.configurationRegistry.registerConfigurations([maxActivationTimeConfiguration]);
+    const disabledExtensionConfiguration: IConfigurationNode = {
+      id: 'preferences.extensions',
+      title: 'Extensions',
+      type: 'object',
+      properties: {
+        [ExtensionLoaderSettings.SectionName + '.' + ExtensionLoaderSettings.Disabled]: {
+          description: 'Disabled extensions',
+          type: 'array',
+          hidden: true,
+        },
+      },
+    };
+
+    this.configurationRegistry.registerConfigurations([maxActivationTimeConfiguration, disabledExtensionConfiguration]);
+  }
+
+  getDisabledExtensionIds(): string[] {
+    return (
+      (this.configurationRegistry
+        .getConfiguration(ExtensionLoaderSettings.SectionName)
+        .get(ExtensionLoaderSettings.Disabled) as string[]) || []
+    );
+  }
```

Comment:
AFAIK seems like we could use the cast on the method as it's relying on generics and use the parameter for default value if missing the value

`get<string[]>(extensionLoaderSettings.Disabled, [])`


```suggestion
  getDisabledExtensionIds(): string[] {
    return (
      (this.configurationRegistry
        .getConfiguration(ExtensionLoaderSettings.SectionName)
        .get<string[]>(ExtensionLoaderSettings.Disabled, [])
    );
  }
```

Reply:
Fixed in new commit.